ci/ray_ci/tester_container.py

```diff
@@ -144,21 +144,21 @@ def _get_test_and_results(
             if path.isfile(log) and file.startswith("bazel_log"):
                 bazel_logs.append(log)
 
-        test_and_results = []
+        tests = {}
         # Parse bazel logs and print test results
         for file in bazel_logs:
             with open(file, "rb") as f:
                 for line in f:
                     event = json.loads(line.decode("utf-8"))
-                    if "testResult" in event:
-                        test_and_results.append(
-                            (
-                                Test.from_bazel_event(event, team),
-                                TestResult.from_bazel_event(event),
-                            )
-                        )
-
-        return test_and_results
+                    if "testResult" not in event:
+                        continue
+                    test = Test.from_bazel_event(event, team)
+                    test_result = TestResult.from_bazel_event(event)
+                    # Obtain only the final test result for a given test in case
+                    # the test is retried.
+                    tests[test.get_name()] = (test, test_result)
```

Comment:
if it is flaky (retried), shouldn't we actually record it as flaky? rather than just dropping the failure data?

Reply:
we can yes, but the state machine doesn't understand that test state yet (it only know how to move when a test passes of failed); add another test state make the machine more complex so let's keep it simple for now until we need that state; if you worry about the lossy nature of this recording, we can keep the raw bazel log job for a while